src/collection/collectionwatcher.cpp

```diff
@@ -897,6 +954,32 @@ void CollectionWatcher::AddChangedSong(const QString &file, const Song &matching
 
 }
 
+#ifdef HAVE_EBUR128
+void CollectionWatcher::PerformEBUR128Analysis(SongList &songs, ScanTransaction *t) {
+
+  if (!song_ebur128_loudness_analysis_) return;
+
+  // FIXME: this should probably be done on thread pool.
+  for (Song &song : songs) {
+    std::optional<EBUR128Measures> loudness_characteristics = EBUR128Analysis::Compute(song);
+    if (loudness_characteristics) {
+      song.set_ebur128_integrated_loudness_lufs(loudness_characteristics->loudness_lufs);
+      song.set_ebur128_loudness_range_lu(loudness_characteristics->range_lu);
+      if (t) {
+        // We have updated the song and we must make note of that,
+        // but we should avoid doing that multiple times,
+        // because that will result in spurious/pointless DB updates.
+        if(!t->new_songs.contains(song)) {
+          t->new_songs << song;
```

Comment:
Maybe move the loudness calculation to the AddChangedSong so it gets logged like the other changes

Reply:
That function is not invoked e.g. on this codepath: https://github.com/strawberrymusicplayer/strawberry/blob/0c1f4750eaa98f7bf45fce3247cdec343c38238a/src/collection/collectionwatcher.cpp#L741-L743

Reply:
It's called in UpdateCueAssociatedSongs() and UpdateNonCueAssociatedSong() before PerformEBUR128Analysis is called.
Changed songs should be logged in AddChangedSong().
So the loudness calculation should probably go in UpdateCueAssociatedSongs() and UpdateNonCueAssociatedSong() before AddChangedSong is called.